src/main/java/org/candlepin/subscriptions/event/EventConflictResolver.java

```diff
@@ -0,0 +1,221 @@
+    }
+
+    rebuildEventWithUpdatedMeasurements(toResolve, measurementUomsToSkip).ifPresent(resolved::add);
+    return resolved;
+  }
+
+  private Map<EventKey, List<EventRecord>> getConflictingEvents(List<Event> incomingEvents) {
+    List<EventKey> eventKeys = incomingEvents.stream().map(EventKey::fromEvent).toList();
+    return eventRecordRepository.findConflictingEvents(eventKeys).stream()
+        .collect(Collectors.groupingBy(e -> EventKey.fromEvent(e.getEvent())));
+  }
+
+  private Optional<EventRecord> buildDeductionEvent(
+      Event toResolve, Measurement measurement, Optional<Double> deduction) {
+    if (deduction.isPresent()) {
+      Double deductedValue = deduction.get();
+      // If we had a conflicting Event, but the measurement value was the same
+      // there's nothing to resolve as they are considered equal. This will
+      // NOT result in a new EventRecord for the measurement.
+      //
+      // The deducted value is expected to be <= 0, so we compare
+      // by adding the incoming value to it to see if the result is 0.
+      if (deductedValue + measurement.getValue() == 0.0) {
+        log.debug("Incoming event measurement is a duplicate. Nothing to resolve.");
+        return Optional.empty();
+      }
+
+      // Create an EventRecord with the -ve value
+      Event deductionEvent = createRecordFrom(toResolve);
+      deductionEvent.setMeasurements(
+          List.of(new Measurement().withUom(measurement.getUom()).withValue(deductedValue)));
+      return Optional.of(new EventRecord(deductionEvent));
+    }
+    return Optional.empty();
+  }
+
+  private Optional<EventRecord> rebuildEventWithUpdatedMeasurements(
+      Event toResolve, Set<String> measurementUomsToSkip) {
+    // Rebuild the Event if there are event measurements to skip, otherwise, create an
+    // EventRecord from the Event we attempted to resolve.
+    EventRecord rebuiltEventRecord = null;
+    if (!measurementUomsToSkip.isEmpty()) {
+      // Remove any measurements that would result in no change, and create an
+      // EventRecord based on the updated incoming event if any measurements need
+      // to be applied.
+      List<Measurement> applicableMeasurements =
+          toResolve.getMeasurements().stream()
+              .filter(m -> !measurementUomsToSkip.contains(m.getUom()))
+              .toList();
+
+      if (log.isDebugEnabled() && !applicableMeasurements.isEmpty()) {
+        log.debug(
+            "Skipping measurements in event since they would result in no change: {}",
+            String.join(",", measurementUomsToSkip));
+      }
+
+      // If there are still applicable measurements, rebuild the event we are resolving to
+      // include only the applicable measurements. Because the event measurement
+      // collection is immutable, we need to copy the Event in order to make the updates.
+      if (!applicableMeasurements.isEmpty()) {
+        Event rebuiltEvent = createRecordFrom(toResolve);
```

Comment:
Same as above, I would like to identify that this event has been altered from the original event with a new field to state that "measurements-altered" or something like this.

Reply:
DONE - next push.